File: codes/ruby/chapter01/answer_A05.rb
# 入力
n, k = gets.split.map(&:to_i)

# 全探索
answer = 0
(1..n).each do |x|
  (1..n).each do |y|
    z = k - x - y  # 白いカードに書かれるべき整数
    if z >= 1 && z <= n
      answer += 1
    end
  end
end

# 出力
puts answer

File: codes/elixir/chapter01/answer_A05.exs
n = String.trim(IO.gets("")) |> String.to_integer
k = String.trim(IO.gets("")) |> String.to_integer

answer = Enum.reduce(1..n, 0, fn x, acc ->
  Enum.reduce(1..n, acc, fn y, acc_inner ->
    z = k - x - y
    if z >= 1 and z <= n do
      acc_inner + 1
    else
      acc_inner
    end
  end)
end)

IO.puts(answer)
